src/Filesystem/File.php

```diff
@@ -556,11 +556,12 @@ public function mime()
         }
         if (function_exists('finfo_open')) {
             $finfo = finfo_open(FILEINFO_MIME);
-            $finfo = finfo_file($finfo, $this->pwd());
             if (!$finfo) {
                 return false;
             }
-            list($type) = explode(';', $finfo);
+            $mime = finfo_file($finfo, $this->pwd());
```

Comment:
OOP style usage looks much clear:

```php
try {
    $finfo = new finfo(FILEINFO_MIME);
    $type = $finfo->file($this->pwd());
    list($type) = explode(';', $return);
    return $type;
} catch (\Exception $e) {
    return false;
}
```

Here I am assuming it throws an exception in case of an error, can't find any docs relating to it.